src/coreComponents/finiteElement/elementFormulations/H1_Hexahedron_Lagrange1_GaussLegendre2.hpp

```diff
@@ -130,21 +132,37 @@ class H1_Hexahedron_Lagrange1_GaussLegendre2 final : public FiniteElementBase
     return numNodes;
   }
 
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  static void getSamplingPointCoordInParentSpace( int const & linearIndex,
+                                                  real64 (& samplingPointCoord)[3] )
+  {
+    int const i0 = linearIndex % numSamplingPointsPerDirection;
+    int const i1 = ( (linearIndex - i0)/numSamplingPointsPerDirection ) % numSamplingPointsPerDirection;
+    int const i2 = ( (linearIndex - i0)/numSamplingPointsPerDirection - i1 ) / numSamplingPointsPerDirection;
+
+    real64 const step = 2 / ( numSamplingPointsPerDirection - 1 );
```

Comment:
Since `constexpr` functions also work for non-`constexpr`, I think Scott Meyers recommends to use `constexpr` as much as possible ( in particular to this kind of `static` functions ).

Reply:
I have tried to label `constexpr` `i0`, `i1` and `i2` but the compiler complains about it.

Reply:
I meant the whole function to be `constexpr`, but then you would have to make it return the result I think.
```c++
static constexpr std::array<real64, 3> getSamplingPointCoordInParentSpace( int const & linearIndex )
```
I don't know if we want to make you do such changes.

Reply:
everything upstream has to be `constexpr` as well. so `linearIndex` would have to be `constexpr`.

Reply:
I don't think the function argument can be `constexpr`. That's, at least, what the compiler says and what I have found online with a quick search. Unless there is a way I am not aware of.